students/Zach Cooper/lesson02/assignment/code/charges_calc.py

```diff
@@ -0,0 +1,149 @@
+    """
+    Parses commands from the commandline
+    """
+    debug_logging_level_help = 'debug logging level \n\
+                        0: No debug messages or log file.\n\
+                        1: Only error messages.'
+
+    parser = argparse.ArgumentParser(description='Process some integers.')
+    parser.add_argument('-i', '--input', help='input JSON file', required=True)
+    parser.add_argument(
+        '-o', '--output', help='ouput JSON file', required=True)
+    parser.add_argument(
+        '-d', '--debug', help=debug_logging_level_help, required=False, default=0)
+    return parser.parse_args()
+
+
+def configure_logging_level(log_level):
+    """
+    Configures the logging
+
+    Sets different log levels for debugger
+    :paramater log_levels: 1(Error), 2(Debug), 3(Warning)
+    :return:
+    """
+    # Breakdown of what will be returned under each level
+    if log_level == 0:
+        return None
+
+    file_handler = logging.FileHandler('charges_calc_log')
+    file_handler.setFormatter(formatter)
+    logger.addHandler(file_handler)
+
+    if log_level == 1:
+        logger.setLevel(logging.ERROR)
+    elif log_level == 2:
+        logger.setLevel(logging.WARNING)
+    elif log_level == 3:
+        logger.setLevel(logging.DEBUG)
+
+    return logger
+
+    # # Could also put levels in a dict
+    # log_level = {'0': logging.CRITICAL,
+    #               '1': logging.ERROR,
+    #               '2': logging.WARNING,
+    #               '3': logging.DEBUG}
+
+
+def load_rentals_file(filename):
+    """
+    Load rental data from source file.
+    :param filename: source file
+    :return: data
+    """
+    logging.debug("Opening file %s", filename)
+    with open(filename) as file:
+        try:
+            data = json.load(file)
+        except Exception as err:
+            logging.error("Error reading input file %s", file)
+            print(f"Error reading the input file: {err}")
+
+    return data
+
+
+def calculate_additional_fields(data):
+    """
+    Calculates additional metrics.
+    # How to deal with Nulls for datetime?
+    :param data: Data from source file.
+    :return:
+    """
+    # Date Field
+    for value in data.values():
+        try:
+            rental_start = datetime.datetime.strptime(value['rental_start'], '%m/%d/%y')
+            rental_end = datetime.datetime.strptime(value['rental_end'], '%m/%d/%y')
+        except ValueError as date_error:
+            logging.error("%s The date format is incorrect.", date_error)
+            print(f"Date format is not correct. Should be m/d/y.")
+            logging.debug(value)
+
+        # Total Days Field
+        value['total_days'] = (rental_end - rental_start).days
+        if value["total_days"] < 0:
+            logging.error("Total days is negative: %s", value)
+            print("Total days should not be negative. Rental start and rental end"
+                  "dates need to be checked.")
+            logging.debug(value)
```

Comment:
Printing is good, but since you aren't stopping the loop, they won't know which value was the problem. I would add to the print statement the value, so they don't have to search through all of the data for the negative numbers.